src/services/auth/auth.service.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { afterEach, describe, expect, it, MockedFunction, vi } from 'vitest'
+import { User, UserApi } from '@/stores/user-manager.store.model'
+import { authenticate, logout, getUserInfo } from './auth.service'
+
+global.fetch = vi.fn()
+
+const mockUserApi: UserApi = {
+  login: 'the_user',
+  mail: 'the_user@test.com',
+  is_admin: true,
+  mymaps_role: 999,
+  role: 'admin',
+  role_id: 1,
+  sn: 'Dupont Marcel',
+  typeUtilisateur: 'prive',
+}
+
+const resultUserInfo: User = {
+  login: 'the_user',
+  mail: 'the_user@test.com',
+  isAdmin: true,
+  mymapsRole: 999,
+  role: 'admin',
+  roleId: 1,
+  name: 'Dupont Marcel',
+  sn: 'Dupont Marcel',
+  typeUtilisateur: 'prive',
+}
+
+describe('Auth service', () => {
+  afterEach(() => {
+    vi.clearAllMocks()
+  })
+
+  describe('#authenticate', () => {
+    it('should authenticate user successfully and return user info', async () => {
+      ;(fetch as MockedFunction<typeof fetch>).mockResolvedValueOnce({
+        ok: true,
+        json: vi.fn().mockResolvedValue(mockUserApi),
+      } as unknown as Response)
```

Comment:
What about creating a helper for this verbose & repeated construct ?
```typescript
const mockFetch = (response: any) => {
  (fetch as MockedFunction<typeof fetch>).mockResolvedValueOnce(response as unknown as Response);
};
```

Reply:
thx for the suggestion, I made some refacto